dask/dataframe/tests/test_indexing.py

```diff
@@ -684,3 +685,37 @@ def test_pandas_nullable_boolean_data_type():
 
     assert_eq(ddf1[ddf2], s1[s2])
     assert_eq(ddf1.loc[ddf2], s1.loc[s2])
+
+
+def test_deterministic_hashing_series():
+    obj = pd.Series([0, 1, 2])
+
+    dask_df = dd.from_pandas(obj, npartitions=1)
+
+    ddf1 = dask_df.loc[0:1]
+    ddf2 = dask_df.loc[0:1]
+
+    assert tokenize(ddf1) == tokenize(ddf2)
+
+    ddf2 = dask_df.loc[0:2]
+    assert tokenize(ddf1) != tokenize(ddf2)
+
+
+def test_deterministic_hashing_dataframe():
+    # Add duplicate column names in order to use _iLocIndexer._iloc path
+    obj = pd.DataFrame([[0, 1, 2, 3], [4, 5, 6, 7]], columns=["a", "b", "c", "c"])
+
+    dask_df = dd.from_pandas(obj, npartitions=1)
+
+    ddf1 = dask_df.loc[0:1, ["a", "c"]]
+    ddf2 = dask_df.loc[0:1, ["a", "c"]]
```

Comment:
It'll be nice to add coverage for the single-column case as well. For example, string "c" instead of ["a", "c"].

Reply:
will do